src/together/filemanager.py

```diff
@@ -260,12 +266,17 @@ def _redirect_error_handler(
                 http_status=response.status_code,
             )
 
-    def redirect_policy(
-        self, url: str, file: Path, purpose: FilePurpose
+    def get_upload_url(
+        self,
+        url: str,
+        file: Path,
+        purpose: FilePurpose,
+        filetype: FileType,
     ) -> Tuple[str, str]:
         data = {
             "purpose": purpose.value,
             "file_name": file.name,
+            "file_type": filetype.value,
```

Comment:
This field will be used by the Files API to process files of different types in a different way (cc @azahed98)